snuba/request_schema.py

```diff
@@ -0,0 +1,132 @@
+from __future__ import annotations
+
+import copy
+import itertools
+import jsonschema
+
+from collections import ChainMap
+from dataclasses import dataclass
+from typing import Any, Mapping
+
+from snuba.query.extensions import QueryExtension
+from snuba.query.query import Query
+from snuba.schemas import Schema, GENERIC_QUERY_SCHEMA
+
+
+@dataclass(frozen=True)
+class Request:
+    query: Query
+    extensions: Mapping[str, Mapping[str, Any]]
+
+    @property
```

Comment:
Should probably deprecate this too, even though it will also warn due to the usage of `Query.get_body`.